pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -63,56 +62,29 @@ public static boolean isEqual(Object[] left, Object[] right) {
     return Arrays.deepEquals(left, right);
   }
 
-  public static boolean isEqual(Collection left, Collection right) {
-    if (left != null && right != null) {
-      return left.toString().equals(right.toString());
+  public static boolean isEqual(Object left, Object right) {
+    if (left != null) {
+      return left.equals(right);
     } else {
-      return left == right;
+      return right == null;
     }
   }
 
+  @SuppressWarnings("unchecked")
   public static boolean isEqualIgnoringOrder(List left, List right) {
     if (left != null && right != null) {
-      ArrayList sortedLeft = new ArrayList(left);
-      ArrayList sortedRight = new ArrayList(right);
+      List sortedLeft = new ArrayList(left);
+      List sortedRight = new ArrayList(right);
       Collections.sort(sortedLeft);
       Collections.sort(sortedRight);
-      return sortedLeft.toString().equals(sortedRight.toString());
-    } else {
-      return left == right;
-    }
-  }
-
-  public static boolean isEqual(Map left, Map right) {
-    if (left != null && right != null) {
-      if (left.size() != right.size()) {
-        return false;
-      }
-      for (Object key : left.keySet()) {
-        if (right.containsKey(key)) {
-          Object leftValue = left.get(key);
-          Object rightValue = right.get(key);
-          if (!isEqual(leftValue, rightValue)) {
-            return false;
-          }
-        } else {
-          return false;
-        }
-      }
-      return true;
+      return sortedLeft.equals(sortedRight);
     } else {
       return left == right;
     }
   }
 
-  public static boolean isEqual(Object left, Object right) {
-    if (left != null)
-      return left.equals(right);
-    else
-      return null == right;
-  }
-
   public static boolean isNullOrNotSameClass(Object left, Object right) {
+    Preconditions.checkNotNull(left);
```

Comment:
Avoid this. 
The method allows right to be null but not left. The semantics are odd and not clear. 
Preconditions throws exception which is not ideal for the library. If you expect left !=null use 'NotNull' annotation